demo/KevinDemo/Stories/Payment/PaymentView.swift

```diff
@@ -0,0 +1,221 @@
+                                }) {
+                                    KevinCountrySelectionRowView(
+                                        title: "kevin_window_main_country_label".localized(),
+                                        countryCode: viewModel.viewState.selectedCountryCode
+                                    )
+                                }
+                            }
+
+                            Group {
+                                Text("kevin_window_main_charity_label".localized())
+                                    .style(.sectionHeader)
+                                    .padding(.top, 28)
+                                    .padding(.bottom, 12)
+
+                                HStack(spacing: 0.0) {
+                                    if viewModel.viewState.isCharityLoading {
+                                        Spacer()
+                                        
+                                        ProgressView()
+                                        
+                                        Spacer()
+                                    } else {
+                                        Spacer(minLength: 0.0)
+
+                                        ForEach(viewModel.viewState.charities, id: \.id) { charity in
+                                            KevinCharityView(
+                                                logoUrlString: charity.logo,
+                                                isSelected: viewModel.viewState.selectedCharity?.id == charity.id,
+                                                onTap: {
+                                                    viewModel.viewState.selectedCharity = charity
+                                                }
+                                            )
+                                            .padding(.trailing, viewModel.viewState.charities.last?.id == charity.id ? 0.0 : 16.0)
+                                        }
+                                        
+                                        Spacer(minLength: 0.0)
+                                    }
+                                }
+                                .frame(height: 51.0)
+                            }
+                            
+                            Group {
+                                Text("kevin_window_main_details_lable".localized())
+                                    .style(.sectionHeader)
+                                    .padding(.top, 28)
+                                    .padding(.bottom, 12)
+
+                                Text("kevin_window_main_email_label".localized())
+                                    .style(.textFieldName)
+                                    .padding(.top, 8.0)
+                                
+                                KevinTextField(
+                                    onChange: {
+                                        viewModel.updateDonateButtonState()
```

Comment:
This seems kinda backwards. ViewModel should observe email state and update donateButtonState.

Reply:
Let's move format changes to the text field itself. Maybe create AmountTextField?